src/box/lua/schema.lua

```diff
@@ -2518,6 +2518,23 @@ base_index_mt.count_luac = function(index, key, opts)
     return internal.count(index.space_id, index.id, itype, key);
 end
 
+-- 1-based iterator-relative index of the first matching tuple. If such a tuple
+-- does not exist, returns the index at which it would be located if existed.
+--
+-- For an existing tuple, if index:index_of(key, {iterator = it}) == N, then
+-- index:pairs(nil, {iterator = it, offset = N - 1}) will start the iterator
+-- from the same tuple as index:pairs(key, {iterator = it})
+base_index_mt.index_of = function(index, key, opts)
```

Comment:
Maybe name it `offset_of` - for consistency with the `offset` parameter? Also, may be make it 0-based so that it can be passed to `pairs` with a decrement?

Reply:
This was the first thig I suggested in the RFC, but it's [been decided](https://www.notion.so/select-offset-N-count-key-log-n-6fead8b03db6409e9d09f56dd58f1006?d=028a099c41a64d249da33a06c526bfef&pvs=4#aa664802eeca45b39f24878ee9bd01a2) to stuck with this solution.

Reply:
I read the comments in the doc. AFAIU the argument for `index_of` was to avoid confusion with `offset`, which is 0-based, while `index_of` was suggested to be 1-based because it's common in the Lua world. But `offset` is already 0-based so I don't understand why we should bother making `index_of` 1-based, thus introducing a different term.

Reply:
Here're the discussion stages:
1. Initially the function was `offset_of` and returned a 0-based value.
2. Then it's been said we better to return 1-based values in the Lua world.
3. Then it's been said it's better to be called `index_of` instead of `offset_of` if it's 1-based.

I aggree with your point but I'm afraid of doing anything against the [RFC](https://www.notion.so/tarantool/select-offset-N-count-key-log-n-6fead8b03db6409e9d09f56dd58f1006), feel free to add a comment there.

Important thing to be noted: the product document required the [`indexof` method returning a 1-based value](https://www.notion.so/tarantool/offset-n-count-key-log-n-4aba908b48674123892474957420c85f?pvs=4#1981fb9b6757424cb961e0c1e574b1bc) (the link to the exact block may not work as expected, see the very last block of the document).

Reply:
I talked to @unera. He says that he suggested the name `index_of` in the original design document because it's named so in Redis. He doesn't mind naming it `offset_of`. He says that making it 0-based would be better because it would be consistent with the `offset` parameter of `select`. Making it 1-based would require the user to subtract 1 from it before using it with `select`.

Another reason I don't like the name `index_of` because `box.index.foo.index_of(...)` looks weird ("index" here is used twice but has completely different meanings).

Reply:
Renamed to `offset_of` and made it return 0-based values. Also added a no-index case to the test suite.